addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -95,6 +102,20 @@ def _compute_state(self):
             elif all(picking.state in ['cancel', 'done'] for picking in batch.picking_ids):
                 batch.state = 'done'
 
+    @api.depends('picking_ids', 'picking_ids.scheduled_date')
+    def _compute_scheduled_date(self):
+        """ scheduled_date = the earliest scheduled_date across all pickings. """
+        self.scheduled_date = False
```

Comment:
@ryv-odoo taught me you can give a default value to min (and max) in case of empty array. Your compute can be shortened into : 
``` python
for batch in self:
    batch.scheduled_date = min(batch.picking_ids.mapped('scheduled_date'), default=False)
```
Also do you need to cast the date as string for some reason ?

Reply:
Also, you can also a filtered on pickings because the scheduled date is not required in the model.
`min(batch.picking_ids.filtered('scheduled_date').mapped('scheduled_date'), default=False)`

Reply:
I think scheduled date is required for pickings (it defaults to now if it's not set at creation and after creation it becomes a mandatory field in the form view). To be safe I added in the filtered anyways for final code of:
`def _compute_scheduled_date(self):`
`     self.scheduled_date = min(self.picking_ids.filtered('scheduled_date').mapped('scheduled_date'), default=False)`